### PR TITLE
Fix Install-Id generation on every launch on UWP

### PR DESCRIPTION
**Pull request by russelarms**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [ ] ~Are tests passing locally?~
* [x] Are the files formatted correctly?
* [ ] ~Did you add unit tests if this modifies the Windows code?~
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Fix the bug that caused **Install-Id** to be generated every time the application was launched.

## Related PRs or issues

[AB#77967](https://msmobilecenter.visualstudio.com/Mobile-Center/_workitems/edit/77967)

### CODE REVIEW

File: Assets/AppCenter/Plugins/AppCenterSDK/Core/UWP/UnityApplicationSettings.cs [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 #if UNITY_WSA_10_0 && ENABLE_IL2CPP && !UNITY_EDITOR
 using System.Collections;
+using System;
```
Comment (MatkovIvan) [inline Assets/AppCenter/Plugins/AppCenterSDK/Core/UWP/UnityApplicationSettings.cs]: Sort

File: Assets/AppCenter/Plugins/AppCenterSDK/Core/UWP/UnityApplicationSettings.cs [hunk 2]
```diff
@@ -26,12 +27,28 @@ public T GetValue<T>(string key, T defaultValue)
                 var found = _current.TryGetValue(key, out result);
                 if (found)
                 {
+                    // This is a workaround for a bug on UWP where Install-Id cannot be read 
+                    // from Unity.PlayerPrefs, because it is saved as System.Guid, but read as a string.
+                    if ((typeof(T) == typeof(Guid?)) && (result is string))
```
Comment (MatkovIvan) [inline Assets/AppCenter/Plugins/AppCenterSDK/Core/UWP/UnityApplicationSettings.cs]: Remove extra `()`

File: Assets/AppCenter/Plugins/AppCenterSDK/Core/UWP/UnityApplicationSettings.cs [hunk 3]
```diff
@@ -26,12 +27,28 @@ public T GetValue<T>(string key, T defaultValue)
...
+                    {
+                        try 
+                        {
+                            object guid = (object)(new Guid((string)result));
```
Comment (MatkovIvan) [inline Assets/AppCenter/Plugins/AppCenterSDK/Core/UWP/UnityApplicationSettings.cs]: Remove temp var or extra cast

File: Assets/AppCenter/Plugins/AppCenterSDK/Core/UWP/UnityApplicationSettings.cs [hunk 4]
```diff
@@ -26,12 +27,28 @@ public T GetValue<T>(string key, T defaultValue)
...
+                            return (T)guid;
+                        }
+                        catch (FormatException)
+                        {
+                            Debug.LogError("Guid cannot be parsed due to a format exception.");
```
Comment (MatkovIvan) [inline Assets/AppCenter/Plugins/AppCenterSDK/Core/UWP/UnityApplicationSettings.cs]: Why the exception message is not printed?

**Review by MatkovIvan — CHANGES_REQUESTED** [review]

File: Assets/AppCenter/Plugins/AppCenterSDK/Core/UWP/UnityApplicationSettings.cs [hunk 5]
```diff
@@ -26,12 +27,28 @@ public T GetValue<T>(string key, T defaultValue)
...
+                            return defaultValue;
+                        }
```
Comment (Jamminroot) [inline Assets/AppCenter/Plugins/AppCenterSDK/Core/UWP/UnityApplicationSettings.cs]:
```suggestion
                        if (Guid.TryParse((string) result, out var guid)) {
                            return guid;
                        }
                       else
                       {
                            Debug.LogError("Guid cannot be parsed due to a format exception.");
                            return defaultValue;
                        }
```

Comment (annakocheshkova) [reply Assets/AppCenter/Plugins/AppCenterSDK/Core/UWP/UnityApplicationSettings.cs]: It won't compile: to make it compile we need a double cast

**Review by Jamminroot — APPROVED** [review]

**Review by MatkovIvan — APPROVED** [review]

**Review by annakocheshkova — APPROVED** [review]

**Review by AnastasiaKubova — APPROVED** [review]

**Review by guperrot — APPROVED** [review]

**Pull request merged**